pkg/generate/config/field.go

```diff
@@ -170,11 +170,11 @@ type FieldConfig struct {
 	// Required indicates whether this field is a required member or not.
 	// This field is used to configure '+kubebuilder:validation:Required' on API object's members.
 	IsRequired *bool `json:"is_required,omitempty"`
-	// IsName indicates the field represents the name/string identifier field
-	// for the resource.  This allows the generator config to override the
-	// default behaviour of considering a field called "Name" or
+	// IsPrimaryKey indicates the field represents the primary name/string
+	// identifier field for the resource.  This allows the generator config to
+	// override the default behaviour of considering a field called "Name" or
 	// "{Resource}Name" or "{Resource}Id" as the "name field" for the resource.
-	IsName bool `json:"is_name"`
+	IsPrimaryKey bool `json:"is_primary_key"`
```

Comment:
My local config has both `is_name` and `is_primary_key` since I had not looked at this code yet. But generator does not throw an error. We should have some validation in the future for extra fields too.

Reply:
Yes, I did a global search for all service controllers using `is_name` and I believe SageMaker is the only example of them. I also agree that the generator should at least warn teams if they have unspecified fields.